### DISCUSSION TITLE
BC setting for 1D horizontal line

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#22958] [Q&A General] [answered]
Dear all,

I have a question: assuming there is a 1D horizontal line. Now I want to set BCs for the left and right boundaries of the system.  If I set two different pressures for each boundary.  I got a result for the pressure and darcy velocity. But I am wondering if the system is automatically set to open or close. If I do not set flux values (i.e., Neumann BC) at boundaries.

Thanks a lot.

**Comment by Joseph-0123** [comment, marked as answer]
@GiudGiud

**Reply by GiudGiud** [reply]
Is this for finite volume or finite element?
In finite volume, you ll need to use the NSFV-specific (outlet) boundary condition on pressure, and it will make the system open on both sides with pressure BCs

Note that we have not worked on inlet+outlet pressure BCs. We ve only done velocity/mass flow inlet + pressure outlet.

**Reply by Joseph-0123** [reply]
We use our finite element application. It's strange that we get open boundaries when we only set pressures for boundary conditions.

**Reply by GiudGiud** [reply]
Which set of kernels are you using? INS? INSAD? THM's?

**Reply by Joseph-0123** [reply]
> Which set of kernels are you using? INS? INSAD? THM's?

We use the  ([HydraulicKernelH](https://github.com/MGK-Lab/tiger/blob/master/src/kernels/TigerHydraulicKernelH.C) and [HydraulicTimeKernelH](https://github.com/MGK-Lab/tiger/blob/master/src/kernels/TigerHydraulicTimeKernelH.C)) in our [application ](https://github.com/MGK-Lab/tiger).

Our kernels are similar to the porous-flow modules.

**Reply by GiudGiud** [reply]
@WilkAndy or @cpgr will know about porous flow. 

A set pressure in fluid flow is generally not a 0 flux condition for mass/momentum

**Comment by friedmud** [comment]
In a 1D system you need to specify at least one Dirichlet condition to create a unique solution.

Generally - for most finite-element formulations - not setting a boundary condition is the same thing as setting a no-flux boundary condition.  The reason why is that when you do the integration by parts to generate a weak form you end up with a flux term on the boundary.  Leaving it out of the simulation is the same thing as setting that term to zero.

**Reply by Joseph-0123** [reply]
> In a 1D system you need to specify at least one Dirichlet condition to create a unique solution.
> 
> Generally - for most finite-element formulations - not setting a boundary condition is the same thing as setting a no-flux boundary condition. The reason why is that when you do the integration by parts to generate a weak form you end up with a flux term on the boundary. Leaving it out of the simulation is the same thing as setting that term to zero.

Thank you. Yes, I set two pressure values for my 1D model boundaries, one for each. However, I did not set flux boundary conditions.
After computation, I got the results with influx and outflux at boundaries which are not set by me. Based on your words, If we donot set the flux boundaries, the system should have boundary with no flux, meaning that is a closed system, right?

**Reply by friedmud** [reply]
Ah - since you set a Dirichlet BC for pressure... that means you can have flux through that boundary.  Think of it this way: you're choosing to constrain the value instead of the gradient... so you will get the specified _value_ but the gradient (which gives you flux) will be whatever it needs to be to satisfy the PDE (i.e. not zero).
